### DISCUSSION TITLE
How to copy the long hash of the commit under the cursor?

### DISCUSSION THREAD

**Discussion by mbeko** [rbong/vim-flog#116] [Q&A] [answered]
I frequently need to find a commit in the history and copy its long hash. I'm aware of `y<C-G>`, but this only copies the short hash.

Currently, I press `Enter` on the commit, switch to the new window, and select the hash from the buffer name displayed on the tab bar with the mouse.

This is pretty tedious because I have to switch to the mouse and aim correctly to highlight the full hash and only the hash. If at least the hash was contained in the commit window, I could copy it with standard normal mode commands using only the keyboard.

Do you have any advice how to reach my goal in an easier way?

**Comment by mbeko** [comment]
I just found out that I can do `:Floggit show <Tab><Tab>` and copy the hash in normal mode from the new window.

That's already an improvement, but I hope there's a quicker way

**Comment by TamaMcGlinn** [comment, marked as answer]
Put this autocmd in your flog augroup, in your vimrc:

```
augroup flog
    autocmd FileType floggraph nno <silent> yh :<C-U>call setreg('+', systemlist(flog#fugitive#GetGitCommand() . ' rev-parse ' . flog#Format("%(h)"))[0])<CR>
augroup END
```

Now `yh` copies the git hash into the '+' register, assuming you need it outside of vim. Otherwise, change that to the empty string for the default (unnamed) register, then just `p` will paste it.

**Reply by mbeko** [reply]
That works great, thanks a lot! :tada:
